packages/babel-plugin-transform-classes/src/transformClass.js

```diff
@@ -413,24 +413,25 @@ export default function transformClass(
     const placement = node.static ? "static" : "instance";
     const methods = classState.methods[placement];
 
-    const { computed } = node;
-    if (computed) methods.hasComputed = true;
-
-    const key = t.toComputedKey(node);
     const descKey = node.kind === "method" ? "value" : node.kind;
+    const key =
+      t.isNumericLiteral(node.key) || t.isBigIntLiteral(node.key)
+        ? t.stringLiteral(String(node.key.value))
+        : t.toComputedKey(node);
 
     let fn = t.toExpression(node);
-    // infer function name
-    if (t.isStringLiteral(key) && node.kind === "method") {
-      fn = nameFunction({ id: key, node: node, scope });
+
+    if (t.isStringLiteral(key)) {
+      // infer function name
+      if (node.kind === "method") {
+        fn = nameFunction({ id: key, node: node, scope });
+      }
```

Comment:
We can also infer the name of getters and setters, but I'll do that in a separate PR since it's not strictly related to this one.